microovn/ovn/ovsdb/ovsdb.go

```diff
@@ -26,3 +41,241 @@ func ExpectedOvsdbSchemaVersion(s *state.State, dbSpec *ovnCmd.OvsdbSpec) (strin
+	}
+
+	results := map[string]string{}
+
+	// Gather expected schema version from every member in the cluster via their API.
+	err = clusterClient.Query(s.Context, true, func(ctx context.Context, c *client.Client) error {
+		clientUrl := c.URL()
+		clientUrlString := clientUrl.String()
+		logger.Debugf("Requesting OVSDB %s schema status from '%s'", dbSpec.FriendlyName, clientUrlString)
+		result, err := microovnClient.GetExpectedOvsdbSchemaVersion(ctx, c, dbSpec)
+		if err != nil {
+			logger.Errorf(
+				"Failed to get OVN %s DB schema status from '%s': %s",
+				dbSpec.FriendlyName,
+				clientUrlString,
+				err,
+			)
+			return err
+		}
+		results[clientUrlString] = result
+		return nil
+	})
+
+	if err != nil {
+		return false, fmt.Errorf("failed to assess overall cluster readiness for OVN cluster DB upgrade: '%s'", err)
+	}
+
+	// Match expected schema version from cluster members against our expected version. Only if all values match,
+	// return "true", indicating that the cluster is ready for the schema upgrade.
+	clusterReady := true
+	for host, status := range results {
+		if status == targetVersion {
+			logger.Infof("Host at '%s' has OVN %s DB ready for upgrade", host, dbSpec.FriendlyName)
+		} else {
+			logger.Infof("Host at '%s' does not have OVN %s DB ready for update", host, dbSpec.FriendlyName)
+			clusterReady = false
+		}
+	}
+
+	return clusterReady, nil
+}
+
+// UpgradeCentralDB checks whether a schema upgrade for the specified DB is required. If so, it triggers
+// a coordinated approach to an OVSDB cluster schema upgrade.
+// The function will wait in loop until every member in the cluster is ready for the schema upgrade. Once the cluster
+// is ready, a designated cluster member will trigger a schema upgrade. Non-designated members will keep looping until,
+// eventually, their expected schema version will match the currently running schema version.
+//
+// Note: This function supports upgrade only for the "central" databases (e.i. OVN_Northbound and OVN_Southbound). It
+//
+//	will return error if other database is specified in the 'dbType' argument.
+//
+// Note2: This function has no effect on a node that does not run OVN "central" services, and it will return silently.
+func UpgradeCentralDB(s *state.State, dbType ovnCmd.OvsdbType) error {
+	dbSpec, err := ovnCmd.NewOvsdbSpec(dbType)
+	if err != nil {
+		return fmt.Errorf("failed create DB specification: '%s'", err.Error())
+	}
+
+	if !dbSpec.IsCentral {
+		return fmt.Errorf("MicroOVN handles upgrades only for Northbound and Southbound clustered DBs")
+	}
+
+	centralActive, err := node.HasServiceActive(s, "central")
+	if err != nil {
+		return fmt.Errorf("failed to query local services: %s", err)
+	}
+
+	if !centralActive {
+		logger.Infof(
+			"OVN %s database won't be upgraded from this node because 'central' service is not active",
+			dbSpec.FriendlyName,
+		)
+		return nil
+	}
+
+	backOffMs := 100
```

Comment:
nit: The number used for back off could be a constant, which would allow the other checks used for time to be multipliers of the constant. If we need to change it, we can change it in one place and the rest of the algorithm will adjust in relation to the constant.